classes/class-wc-connect-payment-methods-store.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+if ( ! class_exists( 'WC_Connect_Payment_Methods_Store' ) ) {
+
+	class WC_Connect_Payment_Methods_Store {
+
+		/**
+		 * @var WC_Connect_Service_Settings_Store
+		 */
+		protected $service_settings_store;
+
+		/**
+		 * @var WC_Connect_API_Client
+		 */
+		protected $api_client;
+
+		/**
+		 * @var WC_Connect_Logger
+		 */
+		protected $logger;
+
+		public function __construct( WC_Connect_Service_Settings_Store $service_settings_store,
+			WC_Connect_API_Client $api_client, WC_Connect_Logger $logger ) {
+
+			$this->service_settings_store = $service_settings_store;
+			$this->api_client             = $api_client;
+			$this->logger                 = $logger;
+
+		}
+
+		public function fetch_payment_methods_from_connect_server() {
+
+			$response_body = $this->api_client->get_payment_methods();
+
+			if ( is_wp_error( $response_body ) ) {
+				$this->logger->log( $response_body, __FUNCTION__ );
+				return;
+			}
+
+			$payment_methods = $this->get_payment_methods_from_response_body( $response_body );
+			if ( is_wp_error( $payment_methods ) ) {
+				$this->logger->log( $payment_methods, __FUNCTION__ );
+				return;
+			}
+
+			$this->logger->log( 'Successfully loaded payment methods from server response.', __FUNCTION__ );
+			$this->update_last_fetch_timestamp();
+
+			// If we made it this far, it is safe to store the object
+			$this->update_payment_methods( $payment_methods );
+
+			// Potentially auto-update the selected method
+			$this->potentially_update_selected_payment_method_from_payment_methods( $payment_methods );
+		}
+
+		protected function potentially_update_selected_payment_method_from_payment_methods( $payment_methods ) {
+			$payment_method_count = count( $payment_methods );
+
+			// No payment methods at all? Clear anything we have stored
+			if ( 0 === $payment_method_count ) {
+				$this->service_settings_store->set_selected_payment_method_id( 0 );
+				return;
+			}
+
+			$payment_method_ids = array();
+			foreach ( (array) $payment_methods as $payment_method ) {
+				$payment_method_id = intval( $payment_method->payment_method_id );
+				if ( 0 !== $payment_method_id ) {
+					$payment_method_ids[] = $payment_method_id;
+				}
+			}
+
+			// Just one? Go ahead and use it
+			if ( 1 === $payment_method_count ) {
+				$this->service_settings_store->set_selected_payment_method_id( $payment_method_ids[ 0 ] );
+				return;
+			}
+
+			// Is the stored method id not in the list? Select the first one
```

Comment:
https://github.com/Automattic/woocommerce-connect-client/issues/447 and https://github.com/Automattic/woocommerce-connect-client/issues/496 should cover us in those cases.

What I'm worried about is that if a merchant has 2 credit cards, we would be essentially choosing one of them for him at random. That doesn't sound very good :(
Since this is a conscious decision, my job as "guy who points things out in PRs" is done, I just want you to consider it carefully :)